File: backend/app/schemas/cash.py
import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel


class CashRecordCreate(BaseModel):
    driver_id: uuid.UUID
    date: date
    record_type: str
    amount: Decimal
    receipt_url: str | None = None
    deposit_location: str | None = None
    reference_number: str | None = None
    notes: str | None = None


class CashRecordUpdate(BaseModel):
    status: str | None = None
    notes: str | None = None


class CashRecordResponse(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    driver_id: uuid.UUID
    date: date
    record_type: str
    amount: Decimal
    receipt_url: str | None
    deposit_location: str | None
    reference_number: str | None
    status: str
    verified_by: uuid.UUID | None
    notes: str | None
    created_at: datetime

    model_config = {"from_attributes": True}


class CashSummaryResponse(BaseModel):
    collected: float
    deposited: float
    outstanding: float
    verified: float

class OutstandingDriverResponse(BaseModel):
    driver_id: str
    driver_name: str
    amount: float
    oldest_date: str

File: frontend/src/types/cash.ts
export interface CashRecord {
  id: string;
  tenant_id: string;
  driver_id: string;
  date: string;
  record_type: string;
  amount: number;
  receipt_url: string | null;
  deposit_location: string | null;
  reference_number: string | null;
  status: string;
  verified_by: string | null;
  notes: string | null;
  created_at: string;
}

export interface CashRecordCreate {
  driver_id: string;
  date: string;
  record_type: string;
  amount: number;
  receipt_url?: string;
  deposit_location?: string;
  reference_number?: string;
  notes?: string;
}

export interface CashRecordUpdate {
  status?: string;
  notes?: string;
}

export interface CashFilters {
  date_from?: string;
  date_to?: string;
  driver_id?: string;
  record_type?: string;
  status?: string;
  page?: number;
  per_page?: number;
}

export interface CashSummary {
  collected: number;
  deposited: number;
  outstanding: number;
  verified: number;
}

export interface OutstandingDriver {
  driver_id: string;
  driver_name: string;
  amount: number;
  oldest_date: string;
}
